src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -51,6 +49,23 @@ function ($key, $value, $isHit) use ($defaultLifetime) {
         );
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function get(string $key, callable $callback, float $beta = null)
+    {
+        // ArrayAdapter works in memory, we don't care about stampede protection
+        if (INF === $beta || (($item = $this->getItem($key)) && !$item->isHit())) {
+            $newItem = ($this->createCacheItem)($key, null, true);
```

Comment:
no need for $newItem, using $item is fine here
which means it should be unconditionally fetched first, before the '"if"

Reply:
What it brings is that it just does not call the getItem() method, saves some CPU for the very same result, is it problematic to avoid loading the potentially original item when conditions for rebuild are met for sure ?

Reply:
yes it is because it breaks the semantics, see #27731

Reply:
Got it, but we are working with an in-memory non-shared implementation here, and actually I think the beta argument should just be ignored in this case, and test should be overriden for this specific implementation. Would it be OK ?

Reply:
Being able to dictate immediate expiry is a feature, so `INF` has to be honored.

Reply:
Yes I see, you won't let the chance to a specific implementation to voluntarily ignore some directive when they don't make sense. It actually makes a lot of sense when you consider only the CacheInterface alone, since it's becoming the only way to explicitly expire an item without having the PSR's or legacy cache interfaces which all includes explicit delete() features.